staging/src/k8s.io/kube-apiextensions-server/test/integration/registration_test.go

```
@@ -0,0 +1,188 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	apiextensionsv1alpha1 "k8s.io/kube-apiextensions-server/pkg/apis/apiextensions/v1alpha1"
+	"k8s.io/kube-apiextensions-server/test/integration/testserver"
+)
+
+func instantiateNoxuResource(t *testing.T, namespace, name string, client *dynamic.ResourceClient, definition *apiextensionsv1alpha1.CustomResource) *unstructured.Unstructured {
+	noxuInstanceToCreate := testserver.NewNoxuInstance(namespace, name)
+	createdNoxuInstance, err := client.Create(noxuInstanceToCreate)
+	if err != nil {
+		t.Logf("%#v", createdNoxuInstance)
+		t.Fatal(err)
```

I prefer to return an error so that I can later test with an invalid one.